plugins/thiefControl/thiefControl.cpp

```diff
@@ -36,33 +36,20 @@ void ThiefControl::Init(const char* /*config*/ )
 
 void ThiefControl::Event (bz_EventData * eventData)
 {
+    if (eventData->eventType != bz_eFlagTransferredEvent)
+      return;
+	
     bz_FlagTransferredEventData_V1 *data = (bz_FlagTransferredEventData_V1 *) eventData;
     const std::string noStealMsg = "Flag dropped. Don't steal from teammates!";
-
-    if (!data)
-        return;
-
-    if (data->eventType != bz_eFlagTransferredEvent)
-        return;
-
-    bz_BasePlayerRecord *playerFrom = bz_getPlayerByIndex(data->fromPlayerID);
-
-    if (!playerFrom)
-        return;
-
-    bz_BasePlayerRecord *playerTo = bz_getPlayerByIndex(data->toPlayerID);
-
-    if (!playerTo)
-    {
-        bz_freePlayerRecord(playerFrom);
-        return;
-    }
+	
+    bz_eTeamType thiefTeam = bz_getPlayerTeam(data->toPlayerID);
+    bz_eTeamType targetTeam = bz_getPlayerTeam(data->fromPlayerID);
```

Comment:
This is the fundamental change, using `bz_getPlayerTeam` instead of `bz_getPlayerByIndex`. While a maintenance burden, the previous code did some explicit error checking for the case where a playerID no long maps to a player record. It may have been pedantic completeness, or it may have been robust. What isn't clear is which, and if this change is equally robust.

Specifically, what does `bz_getPlayerTeam` return in the case where the playerID doesn't map to an existing record?

Beyond that, removing the explicit memory management is a Good Thing™

Reply:
According to "bzfsAPI.cxx", it returns **eNoTeam**, if the _playerID_ does not map to an existing record.

```
BZF_API bz_eTeamType bz_getPlayerTeam( int playerID )
{
    GameKeeper::Player *player = GameKeeper::Player::getPlayerByIndex(playerID);

    if (!player)
        return eNoTeam;

    return convertTeam(player->player.getTeam());
}

Reply:
So for equivalent functionality, you should test both values for `eNoTeam` and early return if either, right?

Reply:
For precisely equivalent functionality, yes.